application/tests/unit/common/models/MethodTest.php

```diff
@@ -57,5 +60,131 @@ public function testGetMaskedValueEmail()
+        $method = Method::createAndSendVerification(
+            $user->id,
+            Method::TYPE_PHONE,
+            $mockPhones[0]['number']
+        );
+
+        $this->assertEquals($mockPhones[0]['code'], $method->verification_code);
+    }
+
+    public function testCreateAndSendVerificationInvalidType()
+    {
+        $this->setExpectedException('\Exception', '', 1461375342);
+        Method::createAndSendVerification(
+            1,
+            'invalid type',
+            'value'
+        );
+    }
+
+    public function testCreateAndSendVerificationInvalidEmail()
+    {
+        $this->setExpectedException('\Exception', '', 1461459797);
+        Method::createAndSendVerification(
+            1,
+            Method::TYPE_EMAIL,
+            'not-a-email'
+        );
+    }
 
+    public function testCreateAndSendVerificationInvalidPhone()
+    {
+        $this->setExpectedException('\Exception', '', 1461375342);
+        Method::createAndSendVerification(
+            1,
+            Method::TYPE_PHONE,
+            'not-a-phone'
+        );
+    }
+
+    public function testValidateAndSetAsVerifiedValid()
+    {
+        $user = $this->users('user1');
+        $mockPhones = include __DIR__ . '/../../../mock/phone/data.php';
+        $method = Method::createAndSendVerification(
+            $user->id,
+            Method::TYPE_PHONE,
+            $mockPhones[0]['number']
+        );
+
+        $this->assertEquals($mockPhones[0]['code'], $method->verification_code);
+
+        $method->validateAndSetAsVerified($mockPhones[0]['code']);
+
+        $this->assertEquals(1, $method->verified);
+        $this->assertNull($method->verification_code);
+        $this->assertNull($method->verification_expires);
+    }
+
+    public function testValidateAndSetAsVerifiedInvalid()
+    {
+        $user = $this->users('user1');
+        $mockPhones = include __DIR__ . '/../../../mock/phone/data.php';
+        $method = Method::createAndSendVerification(
+            $user->id,
+            Method::TYPE_PHONE,
+            $mockPhones[0]['number']
+        );
+
+        $this->assertEquals($mockPhones[0]['code'], $method->verification_code);
+        $this->assertEquals(0, $method->verification_attempts);
+
+        $this->setExpectedException('\Exception', '', 1461442988);
+        $method->validateAndSetAsVerified('asdf1234');
+
+        $this->assertEquals(0, $method->verified);
+        $this->assertEquals(1, $method->verification_attempts);
+        $this->assertNotNull($method->verification_code);
+        $this->assertNotNull($method->verification_expires);
+
+        $this->setExpectedException('\Exception', '', 1461442988);
+        $method->validateAndSetAsVerified('asdf1234');
+        $this->assertEquals(2, $method->verification_attempts);
+    }
+
+    public function testDeleteExpiredUnverifiedMethods()
+    {
+        $method1 = Method::findOne(['uid' => '33333333333333333333333333333333']);
```

Comment:
you could assign the '333...' to a variable, so a code reviewer doesn't need to count to see if they're the same. :-)